mlflow/models/wheeled_model.py

```diff
@@ -0,0 +1,158 @@
+
+        # Check if the model file has `wheels` set to True
+        if model_metadata.__dict__.get(_WHEELS_FOLDER_NAME, None):
+            raise MlflowException("Cannot add wheels to a wheeled model", BAD_REQUEST)
+
+        conda_env = model_metadata.flavors.get(FLAVOR_NAME, {}).get(ENV, None)
+        conda_env_path = os.path.join(local_model_path, conda_env)
+        if conda_env is None:
+            raise MlflowException(
+                "Can not add wheels for model with no conda environment.", BAD_REQUEST
+            )
+        if not os.path.isfile(pip_requirements_path):
+            raise Exception("Can not add wheels for model with no 'requirements.txt'.")
+
+        self._download_wheels(dst_path=wheels_dir, pip_requirements_path=pip_requirements_path)
+
+        # Update requirements.txt with wheels
+        pip_wheels = self._overwrite_pip_requirements_with_wheels(
+            wheels_dir=wheels_dir, pip_requirements_path=pip_requirements_path
+        )
+
+        _mlflow_conda_env(path=conda_env_path, additional_pip_deps=pip_wheels, install_mlflow=False)
+
+        # Update MLModel File
+        mlflow_model = self._update_mlflow_model(
+            mlflow_model=mlflow_model, original_model_metadata=model_metadata
+        )
+        mlflow_model.save(model_metadata_path)
+        return mlflow_model
+
+    def _update_mlflow_model(self, mlflow_model, original_model_metadata):
+        """
+        Modifies the MLModel file to reflect updated information such as the run_id,
+        utc_time_created. Additionally, this also adds `wheels` to the MLModel file to indicate that
+        this is a `wheeled` model.
+        :param  mlflow_model: :py:mod:`mlflow.models.Model` configuration of the newly created
+                                wheeled model
+        :param original_model_file_path: The model metadata stored in the original MLmodel file.
+        """
+
+        run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
+        if mlflow_model is None:
+            mlflow_model = Model(run_id=run_id)
+
+        original_model_metadata.__dict__.update(
+            {k: v for k, v in mlflow_model.__dict__.items() if v}
+        )
+        mlflow_model.__dict__.update(original_model_metadata.__dict__)
+
+        mlflow_model.wheels = _WHEELS_FOLDER_NAME
+        return mlflow_model
+
+    def _download_wheels(self, dst_path, pip_requirements_path):
+        """
+        Downloads all the wheels of the dependencies specified in the requirements.txt file
+        :param dst_path: Path to the directory where the wheels are to be downloaded
+        :param pip_requirements_path: Path to requirements.txt in the model directory
+        """
+        if not os.path.exists(dst_path):
+            os.makedirs(dst_path)
+
+        download_command = (
+            f"python -m pip wheel --only-binary=:all: --wheel-dir={dst_path} -r"
+            f"{pip_requirements_path} --no-cache-dir"
```

Comment:
If we run this method on windows, wheels for windows (e.g. [scikit_learn-1.1.2-cp39-cp39-win_amd64.whl](https://files.pythonhosted.org/packages/45/d1/9dfb055ce2893c309936d01efab048ac8ee89ec7ac006fa1f65ff67edaad/scikit_learn-1.1.2-cp39-cp39-win_amd64.whl) for scikit-learn) will be downloaded and the logged model cannot be served on Mac or Linux. Is this a problem?

Reply:
Hey Haru, great question. We discussed this and decided it is not a problem. We are also storing the platform on which the wheels are built.

Reply:
@arjundc-db Can we document this limitation?